pkg/agent/baker.go

```diff
@@ -266,46 +266,57 @@ func normalizeResourceGroupNameForLabel(resourceGroupName string) string {
 }
 
 func validateAndSetLinuxNodeBootstrappingConfiguration(config *datamodel.NodeBootstrappingConfiguration) {
-	// If using kubelet config file, disable DynamicKubeletConfig feature gate and remove dynamic-config-dir
-	// we should only allow users to configure from API (20201101 and later)
-	dockerShimFlags := []string{
-		"--cni-bin-dir",
-		"--cni-cache-dir",
-		"--cni-conf-dir",
-		"--docker-endpoint",
-		"--image-pull-progress-deadline",
-		"--network-plugin",
-		"--network-plugin-mtu",
+	if config.KubeletConfig == nil {
+		return
 	}
 	profile := config.AgentPoolProfile
-	if config.KubeletConfig != nil {
-		kubeletFlags := config.KubeletConfig
-		delete(kubeletFlags, "--dynamic-config-dir")
-		delete(kubeletFlags, "--non-masquerade-cidr")
-		if profile != nil && profile.KubernetesConfig != nil &&
-			profile.KubernetesConfig.ContainerRuntime != "" &&
-			profile.KubernetesConfig.ContainerRuntime == "containerd" {
-			for _, flag := range dockerShimFlags {
-				delete(kubeletFlags, flag)
-			}
+	kubeletFlags := config.KubeletConfig
+
+	// If using kubelet config file, disable DynamicKubeletConfig feature gate and remove dynamic-config-dir
+	// we should only allow users to configure from API (20201101 and later)
+	delete(kubeletFlags, "--dynamic-config-dir")
+	delete(kubeletFlags, "--non-masquerade-cidr")
```

Comment:
what's the link with `non-masquerade-cidr` ?

Reply:
this is old code that needs to remain, though I couldn't tell you the exact reason why we filter out `--non-masquerade-cidr` - I know for a fact that `--dynamic-config-dir` not being filtered out previously has caused outages